types/durableClient.d.ts

```diff
@@ -1,11 +1,65 @@
-import { FunctionInput, HttpRequest, HttpResponse } from "@azure/functions";
+import {
+    FunctionInput,
+    FunctionOptions,
+    FunctionResult,
+    HttpFunctionOptions,
+    HttpRequest,
+    HttpResponse,
+    HttpResponseInit,
+    InvocationContext,
+    Timer,
+    TimerFunctionOptions,
+} from "@azure/functions";
 import { EntityId, EntityStateResponse } from "./entity";
 import { DurableOrchestrationStatus, OrchestrationRuntimeStatus } from "./orchestration";
 
 export interface DurableClientInput extends FunctionInput {
     type: "durableClient";
 }
 
+/**
+ * Type of a handler function that is triggered by some trigger
+ * and receives a [[DurableClient]] instance as an input.
+ */
+export type DurableClientHandler = (
+    triggerInput: any,
+    durableClient: DurableClient,
+    context: InvocationContext
+) => FunctionResult<any>;
+
+/**
+ * Configures the inputs, outputs, and handler for a Durable Client function.
+ */
+export interface DurableClientOptions extends Omit<FunctionOptions, "handler"> {
+    handler: DurableClientHandler;
+}
+
+export type HttpDurableClientHandler = (
+    request: HttpRequest,
+    client: DurableClient,
+    context: InvocationContext
+) => FunctionResult<HttpResponseInit | HttpResponse>;
+
+/**
+ * Configures options for an HTTP-triggered Durable Client function.
+ */
+export interface HttpDurableClientOptions extends Omit<HttpFunctionOptions, "handler"> {
+    handler: DurableClientHandler;
```

Comment:
I want to make sure people still get intellisense for the `request` trigger input in JavaScript (where people don't have the types). I think we need this to make it happen, but in general can you double check intellisense in JavaScript with your changes?:

```suggestion
    handler: HttpDurableClientHandler;
```